docs/marionette.region.md

```diff
@@ -421,6 +423,28 @@ myRegion.reset();
 
 This can be useful in unit testing your views.
 
+## Check If View Is Being Swapped By Another
+
+The `isSwappingView` method returns if a view is being swapped by another one. It's useful
+inside region lifecycle events / methods.
+
+The example will show an message when the region is empty  
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var EmptyMsgRegion = Mn.Region.extend({
+  onEmpty() {
+    if (!this.isSwappingView()) {
+      this.$el.append('Empty Region');
+    }    
+  }  
+});
+```
+[Live example](https://jsfiddle.net/marionettejs/c1nacq0c/1/)
+
+> `hasView` will not work reliably inside `onEmpty` because will always evaluate to false
```

Comment:
when you show a view.. it first empties the region.  `hasView` is entirely dependable in returning `false` inside of `onEmpty`  It should always return `false` inside of `onEmpty`.  While someone may assume that when a view is being swapped `hasView` would be true.. that is a misconception.  The region does not yet have the view until `onShow` _after_ the `show` in completed.  There is never a use case for `hasView` showing anything other than `false` inside of `onEmpty`  If `hasView` showed true inside of `onEmpty` that would effectively mean that the empty failed.